src/App.tsx

```diff
@@ -1,61 +1,120 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+ import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo, TodoWithUser } from './types/Todo';
+import { User } from './types/User';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId)
+    || null;
+};
+
+const todosWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+const findMaxId = (todos: TodoWithUser[]) => {
+  const ids = todos.map(todo => todo.id);
+
+  return Math.max(...ids);
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>(todosWithUsers);
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasErorId, setHasErorId] = useState(false);
+
+  const handleInputTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+  setTitle(event?.target.value.trimStart());
+  setHasTitleError(false);
+  };
+
+  const handleSelectUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasErorId(false);
+  }
+
+  const handeleResetForm = () => {
+    setTitle('');
+    setHasTitleError(false);
+
+    setUserId(0)
+    setHasErorId(false);
+  }
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasErorId(!userId);
+    setHasTitleError(!title);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodo: TodoWithUser = {
+      id: findMaxId(todos) + 1,
+      title,
+      userId: +userId,
```

Comment:
No need to use +userId, becouse u have number in state:
const [userId, setUserId] = useState(0);